lzy/fs/src/main/java/ai/lzy/fs/SlotsService.java

```diff
@@ -256,68 +304,120 @@ public void statusSlot(LSA.StatusSlotRequest request, StreamObserver<LSA.StatusS
 
         @Override
         public void destroySlot(LSA.DestroySlotRequest request, StreamObserver<LSA.DestroySlotResponse> response) {
+            var idempotencyKey = IdempotencyUtils.getIdempotencyKey(request);
+
+            if (idempotencyKey != null && loadExistingOpResult(operationService, idempotencyKey,
+                LSA.DestroySlotResponse.class, response, "Cannot destroy slot", LOG))
+            {
+                return;
+            }
+
             LOG.info("LzySlotsApi::destroySlot `{}`: {}.",
                 request.getSlotInstance().getSlotUri(), JsonUtils.printSingleLine(request));
 
             final SlotInstance slotInstance = ProtoConverter.fromProto(request.getSlotInstance());
 
-            final LzySlot slot = slotsManager.slot(slotInstance.taskId(), slotInstance.name());
-            if (slot == null) {
-                var msg = "Slot `" + slotInstance.shortDesc() + "` not found.";
-                LOG.warn(msg);
+            var op = Operation.create(agentId, "DestroySlot: " + slotInstance.name(), idempotencyKey, null);
+            var opSnapshot = operationService.registerOperation(op);
+
+            if (op.id().equals(opSnapshot.id())) {
+                final LzySlot slot = slotsManager.slot(slotInstance.taskId(), slotInstance.name());
+                if (slot == null) {
+                    var msg = "Slot `" + slotInstance.shortDesc() + "` not found.";
+                    LOG.warn(msg);
+
+                    operationService.updateResponse(op.id(), LSA.DestroySlotResponse.getDefaultInstance());
+                    response.onNext(LSA.DestroySlotResponse.getDefaultInstance());
+                    response.onCompleted();
+                    // operationService.updateError(...);
+                    // response.onError(Status.NOT_FOUND.withDescription(msg).asException());
```

Comment:
yes, we may answer with NOT_FOUND if we know that such slot has neve been created